### ISSUE TITLE
Generate Spatial Transformation attack 

### ISSUE THREAD

**Issue by DorotaDR** [IBM/adversarial-robustness-toolbox#40]
Hi, 
I am IBM intern.  I found a bug in Spatial Transformation Attack. 

I tried to create an example of using **Spatial Transformation Attack**. After a lot of attemps with various factors I always get  NO **adversarial image (type of None)** without any error messages. 

Part of the code I used:
The image is an image (np.array; shape: (1,224,244,3)), others variables are simple and common. 
```
image = Image.open(image_path)

if image.mode is not "RGB":
    image = image.convert("RGB")

image = image.resize((224, 224))
image = img_to_array(image)
image = np.expand_dims(image, axis=0)
image = imagenet_utils.preprocess_input(image)

model_RN50 = ResNet50(weights="imagenet")

classifier = KerasClassifier((image_norm.min(), image_norm.max()), model=model_RN50)
ST_attacker = SpatialTransformation(classifier, max_translation=10.0,num_translations=5,max_rotation=30.0,num_rotations=3)  
adversarial_image_ST = ST_attacker.generate(image)
type(adversarial_image_ST)    # result: NoneType
```

What are the appropriate parameters for this attack? In the docs and source code I didn't find any information about the range and "units" (% or pixels or sth else) of max_translation & max_rotation. 

**System information:**
* Windows 10
* Python 3.7
* ART  0.6.0
* TensorFlow 1.13.1
* Keras 2.2.4

**Comment by ririnicolae** [comment]
@DorotaDR, thanks for reporting this! We will try to reproduce the issue and get back to you.

**Comment by ririnicolae** [comment]
About the parameters, the values that you use seem valid. `max_translation` should be in the range `[0, 100]`, while `max_rotation` in [0, 180]. I will clarify that in the documentation. However, when you create the `KerasClassifier`, the `(clip_min, clip_max)` values should be the maximum allowed data range, e.g., `(0, 255)` for ResNet50. Can you confirm that those are indeed the values you are using?

**Comment by DorotaDR** [comment]
So I used image - `np.array` with values from 0 to 1. The `clip_values` are also (0,1).  Parameters
`max_translation` is 10 and max_rotation is 30, so all in the range. 

I used the same classifier for others attack (DeepFool) and it works.

**Comment by ririnicolae** [comment]
Thanks for the additional clarifications, we'll look into it. As you said, all parameters look indeed valid.

**Comment by beat-buesser** [comment]
@DorotaDR Could you please show the complete example script including the imports of packages and the definition of image_norm?

**Comment by DorotaDR** [comment]
Yes, sure. 
So I uploaded my notebook on Git. 
https://github.com/DorotaDR/ART-notebooks/blob/master/Resnet50_and_SpatialTr_Attack.ipynb
I hope I have clearly described everything. Thanks for your help.

**Comment by beat-buesser** [comment]
Great, I'll take a look at it.

**Comment by beat-buesser** [comment]
I have discovered two issues:

Issue 1: SpatialTransformation did return None because it was not able to fool the classifier. I have fixed that with #42 . In such cases it now returns the unchanged images.

Issue 2: In your notebook the classifier could not be fooled because the input images are not normalised appropriately for ResNet50, therefore it always returns the (almost) same prediction. Your notebook worked for me by normalising the images in cell 8 with:
 ```
from tensorflow.keras.applications.resnet50 import preprocess_input
image_norm = preprocess_input(image)
```
